> * mediasoup-worker / ci (windows-2025, cl, cl, Release, false, false, false, false, false, 24) (pull_request)

Here:

```
../../../src/RTC/TransportTuple.cpp(159): error C3535: cannot deduce type for 'auto *' from 'std::_Array_iterator<_Ty,37>'
        with
        [
            _Ty=uint8_t
        ]
../../../src/RTC/TransportTuple.cpp(159): error C2440: 'initializing': cannot convert from 'std::_Array_iterator<_Ty,37>' to '<error type> *'
        with
        [
            _Ty=uint8_t
        ]
../../../src/RTC/TransportTuple.cpp(159): note: No user-defined-conversion operator available that can perform this conversion, or the operator cannot be called
[1296/1386] Compiling C++ object mediasoup-worker.exe.p/src_RTC_SimulcastConsumer.cpp.obj
```

--- a/worker/src/RTC/TransportTuple.cpp
+++ b/worker/src/RTC/TransportTuple.cpp
@@ -3,6 +3,7 @@
 
 #include "RTC/TransportTuple.hpp"
 #include "Logger.hpp"
+#include <cstring> // std::memcpy()
 
 namespace RTC
 {

--- a/worker/src/RTC/TransportTuple.cpp
+++ b/worker/src/RTC/TransportTuple.cpp
@@ -3,7 +3,6 @@
 
 #include "RTC/TransportTuple.hpp"
 #include "Logger.hpp"
-#include <array>
 
 namespace RTC
 {
@@ -155,10 +154,8 @@ namespace RTC
 
 		// Maximum buffer length for two IPv6 addresses and ports plus protocol.
 		static constexpr size_t BufferSize = ((16 + 2) * 2) + 1;
-		std::array<uint8_t, BufferSize> buffer{};
-		// NOTE: Windows cannot deduce the type if using 'auto'.
-		// NOLINTNEXTLINE (hicpp-use-auto)
-		std::array<uint8_t, BufferSize>::iterator it = buffer.begin();
+		uint8_t buffer[BufferSize]         = {};
+		size_t idx                         = 0;
 
 		auto appendSockAddr = [&](const sockaddr* addr)
 		{
@@ -168,11 +165,10 @@ namespace RTC
 				const auto* ip      = reinterpret_cast<const uint8_t*>(&in->sin_addr.s_addr);
 				const uint16_t port = ntohs(in->sin_port);
 
-				it  = std::copy(ip, ip + 4, it);
-				*it = (port >> 8) & 0xFF;
-				it++;
-				*it = port & 0xFF;
-				it++;
+				std::memcpy(buffer + idx, ip, 4);
+				idx += 4;
+				buffer[idx++] = (port >> 8) & 0xFF;
+				buffer[idx++] = port & 0xFF;
 			}
 			else if (addr->sa_family == AF_INET6)
 			{
@@ -180,19 +176,18 @@ namespace RTC
 				const auto* ip      = reinterpret_cast<const uint8_t*>(&in6->sin6_addr);
 				const uint16_t port = ntohs(in6->sin6_port);
 
-				it  = std::copy(ip, ip + 16, it);
-				*it = (port >> 8) & 0xFF;
-				it++;
-				*it = port & 0xFF;
-				it++;
+				std::memcpy(buffer + idx, ip, 16);
+				idx += 16;
+				buffer[idx++] = (port >> 8) & 0xFF;
+				buffer[idx++] = port & 0xFF;
 			}
 		};
 
 		appendSockAddr(localSockAddr);
 		appendSockAddr(remoteSockAddr);
 
-		*it = static_cast<uint8_t>(this->protocol);
+		buffer[idx] = static_cast<uint8_t>(this->protocol);
 
-		this->hash = TransportTuple::GenerateFnv1aHash(buffer.data(), buffer.size());
+		this->hash = TransportTuple::GenerateFnv1aHash(buffer, idx + 1);
 	}
 } // namespace RTC

--- a/worker/src/RTC/TransportTuple.cpp
+++ b/worker/src/RTC/TransportTuple.cpp
@@ -156,7 +156,9 @@ namespace RTC
 		// Maximum buffer length for two IPv6 addresses and ports plus protocol.
 		static constexpr size_t BufferSize = ((16 + 2) * 2) + 1;
 		std::array<uint8_t, BufferSize> buffer{};
-		auto* it = buffer.begin();
+		// NOTE: Windows cannot deduce the type if using 'auto'.
+		// NOLINTNEXTLINE (hicpp-use-auto)
+		std::array<uint8_t, BufferSize>::iterator it = buffer.begin();
 
 		auto appendSockAddr = [&](const sockaddr* addr)
 		{
@@ -166,9 +168,11 @@ namespace RTC
 				const auto* ip      = reinterpret_cast<const uint8_t*>(&in->sin_addr.s_addr);
 				const uint16_t port = ntohs(in->sin_port);
 
-				it    = std::copy(ip, ip + 4, it);
-				*it++ = (port >> 8) & 0xFF;
-				*it++ = port & 0xFF;
+				it  = std::copy(ip, ip + 4, it);
+				*it = (port >> 8) & 0xFF;
+				it++;
+				*it = port & 0xFF;
+				it++;
 			}
 			else if (addr->sa_family == AF_INET6)
 			{
@@ -176,9 +180,11 @@ namespace RTC
 				const auto* ip      = reinterpret_cast<const uint8_t*>(&in6->sin6_addr);
 				const uint16_t port = ntohs(in6->sin6_port);
 
-				it    = std::copy(ip, ip + 16, it);
-				*it++ = (port >> 8) & 0xFF;
-				*it++ = port & 0xFF;
+				it  = std::copy(ip, ip + 16, it);
+				*it = (port >> 8) & 0xFF;
+				it++;
+				*it = port & 0xFF;
+				it++;
 			}
 		};
 

